src/lib.rs

```diff
@@ -423,17 +423,21 @@ fn derive_where_internal(input: TokenStream) -> Result<TokenStream> {
 	// Save `Span` before we consume `input` when parsing it.
 	let span = input.span();
 	let item = syn::parse2::<DeriveInput>(input).expect("derive on unparsable item");
+	let cleaned_item = input_without_derive_where_attributes(item.clone());
```

Comment:
I see now how you were able to skip importing `PhantomData`. The issue here is a bit complicated. When the attribute proc macro fails for any reason, the item also gets removed. This has a very bad user experience, because a missing item can break everything, instead of showing a useful error.

- Save input.
- Parse input, on failure, output both the error and the unparsed input.
- Clean input.
- Do our thing, on failure, output both the error and the cleaned input.

I suggest we handle all of this in `derive_where()`, cleaning the item has also be moved there. This will require some pretty big `clone()`s.

Reply:
sounds like a reasonable thing, I already wondered why PhantomData was not used anymore

Reply:
should be working now, but I moved `derive_where_internal` to test as it is no longer used here.

That means we are not testing this with unit tests anymore. But with integration tests still.